src/handlers/canonicalFieldsHandlers.ts

```diff
@@ -1,13 +1,35 @@
+import { logger } from '../logger';
+import { db } from '../database';
+import { isCanonicalFieldArray } from '../types';
+import { ValidationError } from '../errors';
 import type {
   Request,
   Response,
 } from 'express';
 
-function getCanonicalFields(req: Request, res: Response): void {
-  res.status(200)
-    .contentType('application/json')
-    .send('{ "hello": "world"}');
-}
+const getCanonicalFields = (req: Request, res: Response): void => {
+  db.sql('canonicalFields.fetchAll')
+    .then((canonicalFields) => {
+      logger.debug(canonicalFields, 'canonicalFields.fetchAll');
+      const { rows } = canonicalFields;
+      if (isCanonicalFieldArray(rows)) {
```

Comment:
Super nice.
I was about to say "maybe we don't need to validate from our own database" but if we have the function available why not?

Reply:
I hear ya -- and ... it may well be overkill, but one other super nice thing that this is doing is typing `rows` for static analysis purposes.  After this line TypeScript knows that `rows` is a `CanonicalField[]`.  That doesn't mean much now, but it's possible some day we'll want to be able to validate that this handler does, in fact, retrun a `CanonicalField[]` outside of runtime.

Reply:
Oh, yeah, that wasn't clear just from reading this snippet but that is useful and gives me warm fuzzies.